main/src/ca/uwaterloo/flix/language/phase/Weeder.scala

```diff
@@ -575,8 +578,8 @@ object Weeder extends Phase[ParsedAst.Program, WeededAst.Program] {
       val loc = mkSL(sp1, sp2)
       visitExp(exp) map {
         case e => op match {
-          case "!" => WeededAst.Expression.UnaryDeprecated(UnaryOperator.LogicalNot, e, loc)
-          case "not" => WeededAst.Expression.UnaryDeprecated(UnaryOperator.LogicalNot, e, loc)
+          case "!" => WeededAst.Expression.Unary(SemanticOperator.BoolOp.Not, e, loc)
```

Comment:
Actually `!` has already been removed from the parser, so this case cannot occur.

The same goes for the `||` and `&&` cases below.